app/src/main/java/org/pesmypetcare/mypetcare/services/googlecalendar/GoogleCalendarAdapter.java

```diff
@@ -16,47 +18,107 @@
                 + event.getDateTime().getYear() + event.getDescription()).toLowerCase();
         EventData eventData = new EventData(id, pet.getName(), A_REALLY_PRETTY_LOCATION,
                 event.getDescription(), EventData.BLUEBERRY, EMAIL_REMINDER_MINUTES, 0,
                 event.getDateTime().toString(), event.getDateTime().toString());
-        ServiceLocator.getInstance().getGoogleCalendarManagerClient().createEvent(pet.getOwner()
-                .getGoogleCalendarToken(), pet.getOwner().getUsername(), pet.getName(), eventData);
+        registerNewEventLibraryCall(pet, eventData);
+    }
+
+    /**
+     * Method responsible for calling the library to create a new event.
+     * @param pet The pet for which we want to create a new event
+     * @param eventData The data of the new event
+     */
+    private void registerNewEventLibraryCall(Pet pet, EventData eventData) {
+        ExecutorService executorService = Executors.newSingleThreadExecutor();
+        executorService.execute(() -> {
+            try {
+                ServiceLocator.getInstance().getGoogleCalendarManagerClient().createEvent(pet.getOwner()
+                        .getGoogleCalendarToken(), pet.getOwner().getUsername(), pet.getName(), eventData);
+            } catch (MyPetCareException e) {
+                e.printStackTrace();
+            }
+        });
+        executorService.shutdown();
     }
 
     @Override
-    public void deleteEvent(Pet pet, Event event) throws ExecutionException, InterruptedException {
+    public void deleteEvent(Pet pet, Event event) {
         String id = (pet.getName() + event.getDateTime().getDay() + event.getDateTime().getMonth()
                 + event.getDateTime().getYear() + event.getDescription()).toLowerCase();
-        ServiceLocator.getInstance().getGoogleCalendarManagerClient().deleteEvent(pet.getOwner()
-                .getGoogleCalendarToken(), pet.getOwner().getUsername(), pet.getName(), id);
+        ExecutorService executorService = Executors.newSingleThreadExecutor();
+        executorService.execute(() -> {
+            try {
+                ServiceLocator.getInstance().getGoogleCalendarManagerClient().deleteEvent(pet.getOwner()
+                        .getGoogleCalendarToken(), pet.getOwner().getUsername(), pet.getName(), id);
+            } catch (MyPetCareException e) {
+                e.printStackTrace();
+            }
+        });
+        executorService.shutdown();
     }
 
     @Override
-    public void newSecondaryCalendar(Pet pet) throws ExecutionException, InterruptedException {
-        ServiceLocator.getInstance().getGoogleCalendarManagerClient().createSecondaryCalendar(pet.getOwner()
-                .getGoogleCalendarToken(), pet.getOwner().getUsername(), pet.getName());
+    public void newSecondaryCalendar(Pet pet) {
+        ExecutorService executorService = Executors.newSingleThreadExecutor();
+        System.out.println("CALENDAR TOKEN "+ pet.getOwner().getGoogleCalendarToken());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['+' is not preceded with whitespace.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5650471)